### PR TITLE
Fixes for MySQL signatures

### PR DESCRIPTION
**Pull request by nvie**
- Add signature for `raw()` function
- Fix signature of optional `escape()` args
- Fix signature of `typeCast` param
- Fix signature of `getConnection()`

### CODE REVIEW

File: definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js [hunk 1]
```diff
@@ -29,13 +29,15 @@ declare module 'mysql' {
...
 
-  declare type QueryOptions = {
-    sql: string,
-    typeCast?: boolean | (field: Object, next: Function) => any,
-    nestTables?: boolean | string, // string form is a separator used to produce column names
-    values?: Array<mixed>,
-    timeout?: number
-  } | string;
+  declare type QueryOptions =
+    | {
```
Comment (GAntoine) [inline definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js]: Is this prettier doing its thing? If that's the case I'll revert this one change before merging the PR.

**Comment by GAntoine** [comment]
@nvie Thanks for the PR, I have have one question on the review.

File: definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js (hunk 1 above)
Comment (nvie) [reply definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js]: Yeah, I guess so. This just ran whenever I tried to check in something in this repo. Apparently ran by a precommit hook of sorts?

Comment (GAntoine) [reply definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js]: Yeah that happens :/ In that case this LGTM, I'll make that tweak and merge it in tonight.

Comment (nvie) [reply definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js]: Thanks 👍

Comment (nvie) [reply definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js]: Hey @GAntoine! Is there anything blocking this PR to be merged? Happy to help out with any questions.

Comment (nvie) [reply definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js]: Btw, I also just snuck in this little addition while I was at it: 2523b515eb1adf49e2f6812a272fa333fa757be3

Comment (GAntoine) [reply definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js]: Not exactly, I just need to get around to it, but things have been hectic the last few days. If you're able to fix the prettier stuff (tip: commit without running the post-commit hook by commenting it out in package.json) and fix the failing specs, I'll merge it right in.

**Comment by nvie** [comment]
Allright, @GAntoine, I've now excluded the Prettier changes from this PR then. I figured standardizing on Prettier's preferred syntax would be the purpose? Hopefully this makes the PR a little easier to review. I'd recommend running Prettier and never looking back after merging this PR, for the next contributors. Thanks! 👍

**Comment by GAntoine** [comment]
Closed in favour of #1632

**Pull request closed without merging**